Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -0,0 +1,431 @@
+          operation.get_previous(), operation.get_value()));
+      });
+  }
+};
+
+EditableTableView::EditableTableView(
+    std::shared_ptr<TableModel> table, std::shared_ptr<HeaderModel> header,
+    std::shared_ptr<TableFilter> table_filter,
+    std::shared_ptr<CurrentModel> current,
+    std::shared_ptr<SelectionModel> selection, ViewBuilder view_builder,
+    Comparator comparator, QWidget* parent)
+    : TableView(std::make_shared<EditableTableModel>(std::move(table), header),
+        std::make_shared<EditableTableHeaderModel>(header),
+        std::move(table_filter), std::move(current), std::move(selection),
+        std::bind_front(&EditableTableView::make_table_item, this,
+          std::move(view_builder)),
+        std::move(comparator), parent) {
+  auto table_header = static_cast<TableHeader*>(static_cast<Box*>(
+    layout()->itemAt(0)->widget())->get_body()->layout()->
+      itemAt(0)->widget());
+  auto empty_header_item =
+    static_cast<TableHeaderItem*>(table_header->layout()->itemAt(0)->widget());
+  empty_header_item->set_is_resizeable(false);
+  table_header->get_widths()->set(0, scale_width(26));
+  auto& scroll_box =
+    *static_cast<ScrollBox*>(layout()->itemAt(1)->widget());
+  scroll_box.setFocusPolicy(Qt::NoFocus);
+  m_table_body = static_cast<TableBody*>(&scroll_box.get_body());
+  m_table_body->installEventFilter(this);
+  set_style(*this, TABLE_VIEW_STYLE());
+  for(auto i = 0; i < get_table()->get_row_size(); ++i) {
+    m_rows.push({});
+  }
+  for(auto i = 0; i < m_table_body->get_table()->get_row_size(); ++i) {
+    auto row = m_table_body->layout()->itemAt(i)->widget();
+    update_style(*row, [] (auto& style) {
+      style = TABLE_ROW_STYLE(style);
+    });
+    m_rows.set(m_table_body->get_table()->get<int>(i, 0), row);
+  }
+  set_column_cover_mouse_events_transparent();
+  m_operation_connection = get_table()->connect_operation_signal(
+    std::bind_front(&EditableTableView::on_source_table_operation, this));
+}
+
+connection EditableTableView::connect_delete_signal(
+    const DeleteSignal::slot_type& slot) const {
+  return m_delete_signal.connect(slot);
+}
+
+QWidget* EditableTableView::make_table_item(ViewBuilder source_view_builder,
+    const std::shared_ptr<TableModel>& table, int row, int column) {
+  if(column == 0) {
+    if(row < table->get_row_size() - 1) {
+      auto button = make_delete_icon_button();
+      button->setMaximumHeight(scale_height(26));
+      set_style(*button, DELETE_BUTTON_STYLE());
+      match(*button, DeleteButton());
+      button->connect_click_signal([=] {
+        QTimer::singleShot(DELETE_TIMEOUT_MS,
+          std::bind_front(&EditableTableView::delete_current_row, this));
+      });
+      return button;
+    } else {
+      return make_empty_cell();
+    }
+  } else if(column == table->get_column_size() - 1) {
+    return make_empty_cell();
+  } else {
+    return source_view_builder(std::make_shared<RevertColumnTableModel>(table),
```

Comment:
Why is a ```RevertColumnTableModel``` being used instead of passing the ```table``` argument from the constructor?

Reply:
`table` in the callback function `view_builder` is different from `table` in the constructor. `table` in the callback function `view_builder` is the transformed table after sorting and filtering. `table` from the constructor can be passed to it, but the second parameter `row` must be converted to source's `row`.